### PR TITLE
[CI:DOCS] quadlet: adjust container unit documentation

### LINKED ISSUE #17909: Quadlet: discrepancy of container option default values in documentation and code
### Issue Description

While playing around with the new quadlet functionality, I noticed that quadlet uses different default values for several options than stated in the [documentation](https://docs.podman.io/en/v4.4/markdown/podman-systemd.unit.5.html). I don't know if these are bugs in the code or the documentation is not correct.

### Documentation

Defaults to `all` according to documentation.

https://github.com/containers/podman/blob/cb18a33f63de2a25d2352fc6a31e38908ac5b647/docs/source/markdown/podman-systemd.unit.5.md?plain=1#L165-L174

### Code

The provided capabilities are correctly dropped, but there is no default of `all`, if the option `DropCapability` is empty or not provided at all.

https://github.com/containers/podman/blob/cb18a33f63de2a25d2352fc6a31e38908ac5b647/pkg/systemd/quadlet/quadlet.go#L431-L435

### Documentation

This one is really strange. In the option title it says `defaults to no`, but in the description below, it states `If enabled (which is the default)`.

https://github.com/containers/podman/blob/cb18a33f63de2a25d2352fc6a31e38908ac5b647/docs/source/markdown/podman-systemd.unit.5.md?plain=1#L329-L332

### Code

The comment in the code suggests that `NoNewPrivileges` should be enabled by default, but `noNewPrivileges` is set to `false` if the lookup fails.

https://github.com/containers/podman/blob/cb18a33f63de2a25d2352fc6a31e38908ac5b647/pkg/systemd/quadlet/quadlet.go#L393-L397

### Documenation

According to the documentation, the default value of `VolatileTmp` depends on the value of `ReadOnly`.

https://github.com/containers/podman/blob/cb18a33f63de2a25d2352fc6a31e38908ac5b647/docs/source/markdown/podman-systemd.unit.5.md?plain=1#L380-L384

https://github.com/containers/podman/blob/cb18a33f63de2a25d2352fc6a31e38908ac5b647/docs/source/markdown/podman-systemd.unit.5.md?plain=1#L459-L463

### Code

But in the code, the default value for `VolatileTmp` is always `false`. If `volatileTmp` should behave as documented, the default value would be `readOnly` and not `false`.

https://github.com/containers/podman/blob/cb18a33f63de2a25d2352fc6a31e38908ac5b647/pkg/systemd/quadlet/quadlet.go#L443-L457

### Steps to reproduce the issue

Steps to reproduce the issue
1.
2.
3.

### Describe the results you received

Describe the results you received

### Describe the results you expected

Describe the results you expected

### podman info output

```yaml
If you are unable to run podman info for any reason, please provide the podman version, operating system and its version and the architecture you are running.
```

### Podman in a container

No

### Privileged Or Rootless

None

### Upstream Latest Release

Yes

### Additional environment details

Additional environment details

### Additional information

Additional information like issue happens only occasionally or issue happens with a particular architecture or on a particular setting

### PR DESCRIPTION
**Pull request by xduugu**
Fixes #17909

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: docs/source/markdown/podman-systemd.unit.5.md [hunk 1]
```diff
@@ -143,8 +143,7 @@ Description of `[Container]` section are:
 
 ### `AddCapability=`
 
-By default, the container runs with no capabilities (due to DropCapabilities='all'). If any specific
-caps are needed, then add them with this key. For example using `AddCapability=CAP_DAC_OVERRIDE`.
+Add these capabilities, in addition to the default podman capability set, to the container.
```
Comment (TomSweeneyRedHat) [inline docs/source/markdown/podman-systemd.unit.5.md:146]:
```suggestion
Add these capabilities, in addition to the default Podman capability set, to the container.
```

File: docs/source/markdown/podman-systemd.unit.5.md [hunk 2]
```diff
@@ -401,9 +400,7 @@ This is equivalent to the Podman `--pull` option
 
 ### `ReadOnly=` (defaults to `no`)
 
-If enabled, makes image read-only, with /var/tmp, /tmp and /run a tmpfs (unless disabled by `VolatileTmp=no`).r
-
-**NOTE:** Podman automatically copies any content from the image onto the tmpfs
+If enabled, makes image read-only.
```
Comment (TomSweeneyRedHat) [inline docs/source/markdown/podman-systemd.unit.5.md:403]:
```suggestion
If enabled, makes the image read-only.
```

**Comment by rhatdan** [comment]
/approve
/lgtm
Thanks @xduugu 
/hold

**Comment by TomSweeneyRedHat** [comment]
LGTM
Thx @xduugu !

**Pull request merged**